### DISCUSSION TITLE
Кириллица в URL

### DISCUSSION THREAD

**Discussion by an220022** [totumonline/totum-ru-issues-and-discussions#25] [Как это сделать?] [answered]
Как строку, содержащую символы на кириллице перевести в url кодировку?

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Сейчас в Totum нет функции URL-кодирования, но есть прием, который позволяет выполнить функцию написанную на любом языке и размещенную на вашем сервере — [execSSH](https://docs.totum.online/functions#fn-execSSH).

По умолчанию она заблокирована и если вы еще не делали этого ранее — вам нужно будет ее раскоментировать в `Conf.php` в корневой папке установки `Totum`: `/* protected $execSSHOn = true;*/` —> `protected $execSSHOn = true;`

Далее мы можем вызвать функцию `php` из кода через [execSSH](https://docs.totum.online/functions#fn-execSSH):

```totum

=: linkToDataText(title: "URlENC"; text: $ssh)

ssh: execSSH(ssh: str`"php -r 'echo urlencode(" + $text + ");'"`)

text: '"Текст для url-кодирования"'

```

Параметры для него надо передавать в кавычках — поэтому в строке `text` одинарная + двойная кавычка — это важно!

Еще есть вариант сохранить этот код где-нибудь в поле `totum-code` и вызвать его через [exec](https://docs.totum.online/functions#fn-exec) там, где вам нужно. Тогда с учетом всех кавычек код будет таким:

```totum

=: execSSH(ssh: str`"php -r 'echo urlencode(" + $text + ");'"`)

text: strAdd(str: '"'; str: $#text; str: '"')

// в $#text не должно быть никаких кавычек, одинарных или двойных

```
